ocpp-transport-soap/src/main/kotlin/com/izivia/ocpp/http/HttpClient.kt

```diff
@@ -0,0 +1,111 @@
+package com.izivia.ocpp.http
+
+import com.izivia.ocpp.soap.OcppSoapParser
+import com.izivia.ocpp.soap.RequestSoapMessage
+import com.izivia.ocpp.soap.ResponseSoapMessage
+import com.izivia.ocpp.transport.ClientTransport
+import com.izivia.ocpp.transport.OcppCallErrorException
+import org.http4k.client.JavaHttpClient
+import org.http4k.contract.contract
+import org.http4k.contract.div
+import org.http4k.core.*
+import org.http4k.lens.Path
+import org.http4k.server.Http4kServer
+import org.http4k.server.Undertow
+import org.http4k.server.asServer
+import org.slf4j.LoggerFactory
+import java.util.*
+import kotlin.reflect.KClass
+
+class HttpClient(
+    private val path: String,
+    private val port: Int,
```

Comment:
I would put those settings in a single data class, and make it nullable. If not provided, the client would not be usable to receive messages from server, which may is ok in some use cases.